ruby_programming/git/lesson_a_deeper_look_at_git.md

```diff
@@ -42,3 +42,20 @@ This section contains helpful links to other content. It isn't required, so cons
 * Watch [Get Going from Pro Git](http://git-scm.com/video/get-going) if you'd like a refresher of things.
 * Read [Git Cheat Sheet](https://www.atlassian.com/git/tutorials/atlassian-git-cheatsheet) if you need a reference sheet.
 * Watch [Using Rebase & Merge](https://www.youtube.com/watch?v=f1wnYdLEpgI) for an example on how to use both rebase and merge.
+
+
+### Knowledge Check
+This section contains questions for you to check your understanding of this lesson. If you're having trouble answering the questions below on your own, review the material above to find the answer.
+
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Basics-Undoing-Things'>How do you amend a commit message that you just messed up?</a>
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History'>How do you view the history of your most recent commits?</a> 
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell#_switching_branches'>What are two different uses for `$ git checkout`?</a> 
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Basics-Undoing-Things#_unstaging'>How do you undo a recent commit?</a> 
```

Comment:
I'm seeing here how to unstage a commit not how to undo a commit, can you point out to me in that section where undoing a commit is covered?

Reply:
I could not find a very easy to follow explanation in the resources provided in the lesson about undoing commits. I would really like to link  [this](https://stackoverflow.com/questions/927358/how-do-i-undo-the-most-recent-local-commits-in-git#s-prose%20js-post-body) to that question. It is very straightforward to follow for a new fellow trying to learn. What do you think? :D

Reply:
I think [this resource](https://git-scm.com/book/en/v2/Git-Tools-Reset-Demystified) linked in the fourth assignment goes over everything that needs to be gone over with regards to `reset` and undoing commits, do you disagree?